app/scripts/directives/service-definition-edit.js

```diff
@@ -80,11 +83,42 @@
         scope.transformToYamlDocumentFragment = function (editedJson) {
           var yamlFrag = {};
           angular.forEach(editedJson, function(svalue) {
-            yamlFrag[svalue.name] = svalue.value;
+            if (svalue.name === 'environment') {
+              yamlFrag[svalue.name] = parseEnvironmentVars(svalue);
+            } else {
+              yamlFrag[svalue.name] = svalue.value;
+            }
           });
           return yamlFrag;
         };
 
+        var parseEnvironmentVars = function (svalue) {
+          var obj = {};
+          for(var i = 0; i < svalue.value.length; i++) {
```

Comment:
the forEach() was freaking out when the value is something like: 'some:text:with:colons'. So I had to use the regular 'for'.